src/openms/source/FORMAT/PepXMLFile.cpp

```diff
@@ -279,15 +279,26 @@ namespace OpenMS
 
         if (lookup.empty())
         {
-          if (it->metaValueExists("RT_index")) // Setting metaValue "RT_index" in XTandemXMLFile in the case of X! Tandem.
+          if (it->metaValueExists("spectrum_reference"))
+          {
+            scan_index = it->getMetaValue("spectrum_reference");
```

Comment:
I think spectrum reference is a native id in most cases (string) and not just an integer.
See the idXML files in https://github.com/OpenMS/OpenMS/search?q=spectrum_reference&unscoped_q=spectrum_reference
So this could be a bit dangerous and will probably not work in most cases.